src/TraceEvent/TraceEventSession.cs

```diff
@@ -867,7 +867,14 @@ public void Dispose()
                 if (m_StopOnDispose)
                 {
                     m_StopOnDispose = false;
-                    Stop(true);
+
+                    // Only dispose session and source when required.
+                    // For session just attached to check if it's active, we must not call dispose.
+                    // Otherwise, it will caused unexpected stop of trace sessions.
+                    if (m_Create)
```

Comment:
Can you update the comment here to state that we only stop the session when we were the original creator of it and not for cases where we attach?  It looks like this is the original comment, but it's no longer correct.  Thanks.

Reply:
Otherwise, this looks good to me.  Thanks!

Reply:
done. Thanks for reviewing!